mlrun/feature_store/api.py

```diff
@@ -405,7 +405,8 @@ def ingest(
         raise mlrun.errors.MLRunInvalidArgumentError(
             "feature set and source must be specified"
         )
-
+    if featureset is not None:
+        featureset.spec.validate_steps()
```

Comment:
I understand that you've decided to only do this validation here, and not do it again later in the function when processing the `mlrun_context`. I think it's better for completeness to do the validation there as well - I want to make sure you're skipping it on purpose.